service/history/shard/controller_impl.go

```diff
@@ -353,9 +353,15 @@ func (c *ControllerImpl) acquireShards() {
 			c.contextTaggedLogger.Error("Unable to create history shard context", tag.Error(err), tag.OperationFailed, tag.ShardID(shardID))
 			return
 		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), shardIOTimeout)
+		defer cancel()
+		// trust the AssertOwnership will handle shard ownership lost
+		_ = shard.AssertOwnership(ctx)
+
 		// Wait up to 1s for the shard to acquire the rangeid lock.
 		// After 1s we will move on but the shard will continue trying in the background.
-		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+		ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
 		defer cancel()
 		_, _ = shard.GetEngine(ctx)
```

Comment:
how about do it like this? there's no reason to AssertOwnership if it's not loaded yet, and this makes it clearer
```suggestion
		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
		defer cancel()
		_, err = shard.GetEngine(ctx)
		
		if err == nil {
			ctx, cancel := context.WithTimeout(context.Background(), shardIOTimeout)
			defer cancel()
			// trust the AssertOwnership will handle shard ownership lost
			_ = shard.AssertOwnership(ctx)
		}
```